crowbar_framework/app/controllers/installer/upgrades_controller.rb

```diff
@@ -223,6 +228,7 @@ def save_and_restore
         flash[:info] = t("installer.upgrades.restore.multiple_restore")
         true
       else
+        Crowbar::Backup::Restore.purge
```

Comment:
> Can't this cause a problem?

no as it is double checked if the file exists before deleting

>  Or is the restore steps file remove on failure anyway?

https://github.com/crowbar/crowbar-core/blob/master/crowbar_framework/lib/crowbar/backup/restore.rb#L44

but i see your confusion, I think we could bring together the `cleanup` method with `purge` somehow, but I'd prefer to do that afterwards with a better overview when all related PRs are merged